dask/array/overlap.py

```diff
@@ -534,7 +609,7 @@ def overlap(x, depth, boundary):
     new_chunks = tuple(
         ensure_minimum_chunksize(size, c) for size, c in zip(depths, x.chunks)
     )
-    x1 = x.rechunk(new_chunks)
+    x1 = x.rechunk(new_chunks)  # this is a no-op if x.chunks == new_chunks
```

Comment:
It's good that there is a no-op path for this, since implementing `rechunk` as a HLG layer would greatly increase the scope here.